File: email-sender-archetype/src/main/resources/archetype-resources/src/main/java/__pluginName__/__pluginName__EmailProviderDescriptor.java
package ${package}.${pluginName};

import se.curity.identityserver.sdk.email.Emailer;
import se.curity.identityserver.sdk.plugin.descriptor.EmailProviderPluginDescriptor;

public final class ${pluginName}EmailProviderDescriptor implements EmailProviderPluginDescriptor<${pluginName}EmailProviderConfig>
{
    @Override
    public Class<? extends Emailer> getEmailSenderType()
    {
        return ${pluginName}EmailSender.class;
    }

    @Override
    public String getPluginImplementationType()
    {
        return "${pluginName.toLowerCase()}";
    }

    @Override
    public Class<? extends ${pluginName}EmailProviderConfig> getConfigurationType()
    {
        return ${pluginName}EmailProviderConfig.class;
    }    
}

File: kotlin-email-sender-archetype/src/main/resources/archetype-resources/src/main/kotlin/__pluginName__/__pluginName__EmailProviderDescriptor.kt
package ${package}.${pluginName}

import se.curity.identityserver.sdk.plugin.descriptor.EmailProviderPluginDescriptor

class ${pluginName}EmailProviderDescriptor: EmailProviderPluginDescriptor<${pluginName}EmailProviderConfig>
{
    override fun getEmailSenderType() = ${pluginName}EmailSender::class.java

    override fun getPluginImplementationType() = "${pluginName.toLowerCase()}"

    override fun getConfigurationType() = ${pluginName}EmailProviderConfig::class.java
}
